### PR TITLE
Print license information in JSON format

### PR DESCRIPTION
**Pull request by aosoriodw**
Changes in this PR are made to support the generation of all the licenses used by dependencies, but without including libraries or versions.

A new command line option `output-type` was added to control the output of text format. It can take two options:
markdown: Preserves the current behaviour and it's the default
json: Prints license information in JSON format and can be parsed with other tools to extract required information.

In addition to these changes I've added tests in a few areas that could have been impacted by the change:
- Generation of tar files: List tar contents
- Markdown output: Verify output

There was also an error in the makefile with the wrong '.PHONY' annotation and this was fixed.

### CODE REVIEW

**Comment by aosoriodw** [comment]
@LukeShu @datawire/infrastructure please take a look

File: main.go
Comment (aosoriodw) [reply main.go:407]: Yes, that's really long.

I think the function is taking too many parameters and with some refactoring maybe it can be made cleaner but I didn't want to spend too much time doing that (at least not for now).

I'll split the line so it's easier to read

File: main.go
Comment (AidanWenzel) [inline main.go:408]: These lines still run over the github window size - makes it a bit harder to review. I don't think this is a blocking problem but we should be aware of it and maybe seek out an autoformatting system to use in the future. Everything else LGTM.

**Review by AidanWenzel — APPROVED** [review]

File: main.go
Comment (AidanWenzel) [reply main.go:408]: Worth noting I'm not comfortable on the language/project so I would still wait for @LukeShu to review before merging if he's available.

Comment (aosoriodw) [reply main.go:408]: > These lines still run over the github window size - makes it a bit harder to review. I don't think this is a blocking problem but we should be aware of it and maybe seek out an autoformatting system to use in the future. Everything else LGTM.

Yes, github displays like 95 characters. I went with what I could see on my display but I agree it depends how you look at the PR.

@LukeShu any suggestions about this? Is there a good reference we can use company-wide for Go code style?
Found some guidelines Uber published: https://github.com/uber-go/guide/blob/master/style.md#style but it's more than just style. Maybe we want to adjust this to how we work.

BTW, that guide suggests a limit of 99 characters per line.

File: README.md
Comment (khussey) [inline README.md:97]: wht -> when?

File: README.md
Comment (khussey) [inline README.md:105]: markdown -> json?

File: README.md
Comment (khussey) [inline README.md:97]: wht -> when?

File: README.md
Comment (aosoriodw) [reply README.md]: Fixed

File: README.md
Comment (aosoriodw) [reply README.md]: Fixed

**Review by khussey — APPROVED** [review]
README looks better, thanks.

**Review by rhs — CHANGES_REQUESTED** [review]
I'm gonna mark this "Request Changes" just so that you can consider my comment. As I say in the comment, my suggestion might not be warranted, but worth considering.

File: main_test.go
Comment (aosoriodw) [reply main_test.go]: This makes a lot of sense. It's not a big change for validating JSON output and it has the good side effect of printing a better diff when things fail
```
        	            	-   Version: (string) (len=8) "v1.17.32",
        	            	+   Version: (string) (len=7) "v1.17.3",
```

It's more difficult for markdown output (which I agree is a fragile test), but I will leave those as is for now.

**Pull request merged**
